File: solutions/2026-04-23/solution.java
// Valid Palindrome
// Difficulty: Easy
// URL: https://leetcode.com/problems/valid-palindrome/

public boolean canPermutePalindrome(String s) {
  s = s.replaceAll(" ", "").toLowerCase();
  int count = 0;
  for (char c : s.toCharArray()) {
    if (s.indexOf(c) == s.lastIndexOf(c)) {
      count++;
    }
  }
  return s.length() % 2 == 0 || count <= 1;
}

File: solutions/2026-04-23/solution.py
# Valid Palindrome
# Difficulty: Easy
# URL: https://leetcode.com/problems/valid-palindrome/

def canPermutePalindrome(s: str) -> bool:
  return len(s) % 2 == 0 or
         len(s.replace(' ', '').lower()) -
         len(set(s.replace(' ', '').lower())) <= 1